applications: serial_lte_modem: SLM server fixes. (#12590)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/TCPUDP_AT_commands.rst

* The ``<error>`` value is an integer.
  It represents the error value according to the standard POSIX *errno*.

Unsolicited notification
~~~~~~~~~~~~~~~~~~~~~~~~

::

   #XTCPSVR: "<peer ip>","connected"

This is emitted when a new connection has been created to the server.

* The ``<peer_ip>`` value is the IPv4 or IPv6 address of the peer.

.. tcpsvr_disconnect_notif_start

::

   #XTCPSVR: <error>,"disconnected"

This is emitted when the client has been disconnected.

* The ``<error>`` value is an integer.
  It is either ``0`` when the client is disconnected normally, or an *-errno* code.

> .. tcpsvr_disconnect_notif_end

Reviewer: ```suggestion

.. tcpsvr_disconnect_notif_end
```
Breaks doc build.

* The ``<error>`` value is an integer.
  It represents the error value according to the standard POSIX *errno*.

Unsolicited notification
~~~~~~~~~~~~~~~~~~~~~~~~

::

   #XTCPSVR: "<peer ip>","connected"

>    This is emitted when a new connection has been created to the server.

Reviewer: Is this meant to be in the code-block? Same for Line 82.

* The ``<peer_ip>`` value is the IPv4 or IPv6 address of the peer.

Unsolicited notification
~~~~~~~~~~~~~~~~~~~~~~~~

.. tcpsvr_status_notif_start

::

   #XTCPSVR: <error>,"disconnected"

   This is emitted when the client has been disconnected.

> * The ``<error>`` value is an integer of 0, for normal closure and *-errno* for erroneous closure.

Reviewer: ```suggestion
* The ``<error>`` value is an integer.
  It is either ``0`` when the server closed normally, or an *-errno* code.
```
Check that I got the meaning right.

Reviewer: So this would be more correct, then:
```suggestion
* The ``<error>`` value is an integer.
  It is either ``0`` when the client disconnects normally, or an *-errno* code.
```

Author: The disconnected comes in both cases when the server closes the connection via hangup or when the client closes the connection.

So maybe like this:

  It is either ``0`` when the client is disconnected normally, or an *-errno* code.

Reviewer: I'd say the first one is better, just to make sure that the timing is clear.

::

   #XTCPDATA: <size>
   <data>

> * The ``<size>`` parameter is the size of the received data. This notification comes only when SLM is not operating in the ``slm_data_mode``.

Reviewer: ```suggestion
* The ``<size>`` parameter is an integer that indicates the size of the received data.
  This notification comes only when SLM is not operating in :ref:`data mode <slm_data_mode>`.
```

::

   #XUDPDATA: <size>
   <data>

> * The ``<size>`` parameter is the size of the received data. This notification comes only when SLM is not operating in the ``slm_data_mode``.

Reviewer: ```suggestion
* The ``<size>`` parameter is an integer that indicates the size of the received data.
  This notification comes only when SLM is not operating in :ref:`data mode <slm_data_mode>`.
```
